### PR TITLE
Shared roles reuse

### PR DESCRIPTION
**Pull request by suvajit-sarkar**
**Changelog**

- Update helm lint build directory check (helm_lint)
- Update  ambassador wait check
- Update  aws cli bin directory check
- Update flux wait check
- Update ha-proxy wait check
- Make the shared check/helm_component for modular
- fix a bug in Quorum, causing deployment to fail
- remove quorum check unused code

 

**Reviewed by**
@lakshyakumar 
 

**Linked issue**
#591

### CODE REVIEW

File: platforms/shared/configuration/roles/check/helm_component/tasks/main.yaml [hunk 1]
```diff
@@ -30,41 +30,20 @@
...
   k8s_info:
     kind: "Pod"
     namespace: "{{ namespace }}"
-    kubeconfig: "{{ kubernetes.config_file }}"
-    context: "{{ kubernetes.context }}"
-    label_selectors:
-      - app = {{ component_name }}
+    kubeconfig: "{{ kubeconfig | default(kubernetes.config_file) }}"
+    context: "{{ context | default(kubernetes.context) }}"
+    label_selectors: "{{ label_selectors }}"
```
Comment (sownak) [inline platforms/shared/configuration/roles/check/helm_component/tasks/main.yaml:42]: this line is not needed as it is not transforming anything. Instead add a comment for this role and update the readme file as to how the label_selectors should be populated when calling this role.

File: platforms/shared/configuration/roles/helm_lint/tasks/main.yaml [hunk 2]
```diff
@@ -14,10 +14,11 @@
 
 # create temp test directory
 - name: "Check or create test directory"
-  file:
+  include_role:
+    name: "../../shared/configuration/roles/check/setup"
```
Comment (sownak) [inline platforms/shared/configuration/roles/helm_lint/tasks/main.yaml:18]: role name can be only check/setup ; they are in the same dir

Comment (suvajit-sarkar) [reply platforms/shared/configuration/roles/helm_lint/tasks/main.yaml:18]: yes even i thought so but looks like it takes the reference from the playbook which ran it

File: platforms/shared/configuration/roles/check/helm_component/tasks/main.yaml (hunk 1 above)
Comment (suvajit-sarkar) [reply platforms/shared/configuration/roles/check/helm_component/tasks/main.yaml:42]: Yeah got it, make sense.. will update that

File: platforms/shared/configuration/roles/helm_lint/tasks/main.yaml (hunk 2 above)
Comment (sownak) [reply platforms/shared/configuration/roles/helm_lint/tasks/main.yaml:18]: ok, then this is fine.

**Pull request merged**